### PR TITLE
New. CheckEmailExist. Visualization of email address verification when filling in the field

### PR DESCRIPTION
**Pull request by AntonV1211**
https://doboard.com/1/task/9864

### CODE REVIEW

File: js/src/apbct-public--2--public.js [hunk 1]
```diff
@@ -165,6 +166,151 @@ function checkEmail(e) {
...
+                            ctCheckedEmailsExist[currentEmail] = {
+                                'result': result.result,
+                                'timestamp': Date.now() / 1000 |0,
+                            };
+                            if (result.result == 'EXISTS') {
+                                viewCheckEmailExist(e, 'good_email', result.text_result);
+                            } else {
+                                viewCheckEmailExist(e, 'bad_email', result.text_result);
+                            }
+                            // ctSetCookie('ct_checked_emails_exist', JSON.stringify(ctCheckedEmailsExist));
```
Comment (Glomberg) [inline js/src/apbct-public--2--public.js:194]: Have to be removed this commented code?

File: js/src/apbct-public--2--public.js [hunk 2]
```diff
@@ -165,6 +166,151 @@ function checkEmail(e) {
...
+    if (currentEmail && !(currentEmail in ctCheckedEmailsExist)) {
+        viewCheckEmailExist(e, 'load');
+        // Using REST API handler
+        if ( ctPublicFunctions.data__ajax_type === 'rest' ) {
+            apbct_public_sendREST(
+                'check_email_exist_post',
+                {
+                    method: 'POST',
+                    data: {'email': currentEmail},
+                    callback: function(result) {
```
Comment (Glomberg) [inline js/src/apbct-public--2--public.js:183]: This callback function is duplicated into the common ajax handler too. Please, extract it.

File: lib/Cleantalk/ApbctWP/API.php [hunk 3]
```diff
@@ -46,6 +46,27 @@ public static function methodPrivateListAddSfwWl($user_token, $service_id, $ip)
...
+     * GET method for getting information
+     * whether an email exists
+     * @param $email
+     * @param $api_key
+     *
+     * @return array|bool|mixed
+     */
+    public static function methodEmailCheckExist($email, $api_key = '')
+    {
+        static::$url = static::URL;
```
Comment (Glomberg) [inline lib/Cleantalk/ApbctWP/API.php:59]: Why this code is here? What it to do?

File: lib/Cleantalk/Common/API.php [hunk 4]
```diff
@@ -23,6 +23,7 @@ class API
     const URL = 'https://api.cleantalk.org';
     const AGENT = 'ct-api-3.2';
 
+    protected static $url;
```
Comment (Glomberg) [inline lib/Cleantalk/Common/API.php:26]: It seems no matters to be added this property.

File: inc/cleantalk-updater.php [hunk 5]
```diff
@@ -1275,3 +1275,11 @@ function apbct_update_to_6_17_2()
     $cron->removeTask('clear_old_session_data');
     $cron->addTask('clear_old_session_data', 'apbct_cron_clear_old_session_data', 86400);
 }
+
+function apbct_update_to_6_38()
```
Comment (Glomberg) [inline inc/cleantalk-updater.php:1279]: As I remember, no need to write updater for the new options. Not is?

**Review by Glomberg — CHANGES_REQUESTED** [review]

File: js/src/apbct-public--2--public.js (hunk 1 above)
Comment (AntonV1211) [reply js/src/apbct-public--2--public.js]: Fixed in new commits

File: inc/cleantalk-updater.php (hunk 5 above)
Comment (AntonV1211) [reply inc/cleantalk-updater.php]: Fixed in new commits

File: lib/Cleantalk/Common/API.php (hunk 4 above)
Comment (AntonV1211) [reply lib/Cleantalk/Common/API.php]: Fixed in new commits

File: js/src/apbct-public--2--public.js (hunk 2 above)
Comment (AntonV1211) [reply js/src/apbct-public--2--public.js:184]: Fixed in new commits

**Review by Glomberg — APPROVED** [review]

**Review by alexandergull — APPROVED** [review]

**Review by Glomberg — APPROVED** [review]

**Pull request merged**